src/Workspaces/Core/MSBuild.BuildHost/Program.cs

```diff
@@ -31,6 +34,16 @@ internal static async Task Main(string[] args)
 
         var logger = new BuildHostLogger(Console.Error);
 
+        try
+        {
+            CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo(locale);
+        }
+        catch (CultureNotFoundException)
```

Comment:
I absolutely expect this check is needed, since the locale here might cross between .NET Core and .NET Framework (and in both directions), so best to be careful here.